tests/asahirov/test_homework_4.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+
+def test_collections_sort():
+    # sort data (same for min max)
+    data = [{"age": 16, "name": "John", "sex": "M"},
+            {"age": 34, "name": "Marry", "sex": "F"},
+            {"age": 25, "name": "Mathew", "sex": "M"}]
+
+    sorted_list = sorted(data, key=lambda i: i["age"])
+    assert sorted_list[-1]["age"] == 34
+
+    sorted_list_reverse = sorted(data, key=lambda i: i["age"], reverse=True)
+    assert sorted_list_reverse[-1]["age"] == 16
+
+
+def test_collections_sort_multiple():
+    data = [{"age": 16, "name": "John", "sex": "M"},
+            {"age": 34, "name": "Marry", "sex": "F"},
+            {"age": 25, "name": "Mathew", "sex": "M"}]
+
+    sorted_list = sorted(data, key=lambda i: (i["sex"] == "M", i["age"]))
+    assert sorted_list[0]["name"] == "Marry"
+    assert sorted_list[-1]["name"] == "Mathew"
+
+
+def test_comprehensions_dict():
+    some_dict = {"a": 1, "b": 2, "c": 3, "d": 4, "e": 5}
+
+    updated_dict = [i[0] for i in some_dict.items() if i[1] % 2 != 0]
```

Comment:
also remember we do not have strong guaranttes about  keys order, so better to sort before assertion


    updated_dict = [k for k, v in some_dict.items() if v % 2 != 0]

    updated_dict = [k for (k, v) in some_dict.items() if v % 2 != 0]

Reply:
updated